### ISSUE TITLE
Set encoding in console to UTF8

### ISSUE THREAD

**Issue by nohwnd** [microsoft/testfx#2491]
## Describe the bug

Frameworks or users might use utf8 characters in their output messages. Set console encoding to utf8 (as we do in vstest.console, and as xunit.console does).  https://github.com/microsoft/vstest/issues/4605

Optionally put this under environment variable feature flag.

## Steps To Reproduce

![image](https://github.com/microsoft/testfx/assets/5735905/41eb5857-df2e-415d-8a08-877ac1761ba8)

```csproj
<!-- file mstest108.csproj -->
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

    <IsPackable>false</IsPackable>
    <IsTestProject>true</IsTestProject>
    <EnableMSTestRunner>true</EnableMSTestRunner>
    <OutputType>Exe</OutputType>
    <GenerateTestingPlatformEntryPoint>false</GenerateTestingPlatformEntryPoint>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MSTest" Version="3.2.0" />
    <PackageReference Include="Microsoft.Testing.Extensions.CodeCoverage" Version="17.10.1" />
  </ItemGroup>

  <ItemGroup>
    <Using Include="Microsoft.VisualStudio.TestTools.UnitTesting" />
  </ItemGroup>

</Project>

```

```cs
// file MSTestSettings.cs
[assembly: Parallelize(Scope = ExecutionScope.MethodLevel)]

```

```cs
// file UnitTest1.cs
namespace mstest108;

using Microsoft.Testing.Platform;
using Microsoft.Testing;
using Microsoft.Testing.Framework;
using Microsoft.Testing.Platform.Builder;

[TestClass]
public class UnitTest1
{
    [TestMethod]
    public void TestMethod1()
    {
        Console.OutputEncoding = System.Text.Encoding.UTF8;
        Assert.AreEqual("a", "🤨😘");
    }
}


public static class Program {
    public static async Task<int> Main(string[] args)  {

        Console.OutputEncoding = System.Text.Encoding.UTF8;
                ITestApplicationBuilder builder = await TestApplication.CreateBuilderAsync(args);
        Microsoft.Testing.Platform.MSBuild.TestingPlatformBuilderHook.AddExtensions(builder, args);
        Microsoft.Testing.Extensions.Telemetry.TestingPlatformBuilderHook.AddExtensions(builder, args);
        Microsoft.VisualStudio.TestTools.UnitTesting.TestingPlatformBuilderHook.AddExtensions(builder, args);
        Microsoft.Testing.Extensions.CodeCoverage.TestingPlatformBuilderHook.AddExtensions(builder, args);
        using ITestApplication app = await builder.BuildAsync();
        return await app.RunAsync();
    }
}
```
[mstest108.zip](https://github.com/microsoft/testfx/files/14460219/mstest108.zip)

## Expected behavior

I can see the smileys.

## Actual behavior

I see ??

**Comment by Evangelink** [comment]
> Another idea can be if we think that's a good "default" for our default console "set by default" utf-8 and provide some "command line args" to change it to something else like '--console-output-encoding xxx'

Haven't put much thoughts but this seems the best. A "good" default and an easy way to change it.

**Comment by nohwnd** [comment]
I don't think this can be purely testing framework concern. The encoding needs to align with msbuild integration for dotnet test, if we want to transport the standard output data correctly, otherwise we are losing encoding and information. This is not so painful for english speaking countries, where utf-8 is mostly used in form of emojis that are rarely used in test names and output, but without this we will be breaking chinese and other characters.

**Comment by MarcoRossignoli** [comment]
> The encoding needs to align with msbuild integration for dotnet test, if we want to transport the standard output data correctly, otherwise we are losing encoding and information. 

Yep, this needs to be followed also for "custom" possible output display in case.

**Comment by Evangelink** [comment]
> The encoding needs to align with msbuild integration for dotnet test

Seems like having an option to change the encoding and having `dotnet test` logic calling this option based on current msbuild setting is the logical split. Platform allows to change console encoding and `dotnet test` that is aware of MSBuild sets the value.

**Comment by Youssef1313** [comment]
I think with the new `dotnet test`, there is nothing on our side to do. `dotnet(.exe)` writes to its own console what it received via pipes. If the `dotnet(.exe)` sets its console to use utf8, we are all good. Similar thing probably for `dotnet run` as well. It might still be a concern when running the test executable directly, but I think keeping the defaults as-is and providing an option like `--console-output-encoding` is best. Keep in mind that `Console.OutputEncoding` is unsupported on some platforms (e.g, Android, iOS, tvOS).

**Comment by nohwnd** [comment]
👍 since it is just me complaining.
